pkg/kv/kvserver/allocator/storepool/store_pool.go

```diff
@@ -198,6 +203,14 @@ type StoreDetail struct {
 	// LastAvailable is set when it's detected that a store was available,
 	// i.e. we got a liveness heartbeat.
 	LastAvailable time.Time
+	// storeHealthTracker tracks the store health over the last
+	// IOOverloadTrackedRetention period, which is 10 minutes. This serves to
+	// exclude stores based on historical information and not just
+	// point-in-time information.
+	storeHealthTracker struct {
```

Comment:
[nit, throughout, feel free to ignore] "Store health" seems vague. Why not call it what is is -- storeIOOverloadTracker?